crates/driver/src/boundary/quote.rs

```diff
@@ -0,0 +1,77 @@
+use {
+    crate::{
+        boundary,
+        domain::{competition::solution, eth},
+        infra::blockchain::Ethereum,
+    },
+    shared::http_solver::model::InternalizationStrategy,
+    solver::{
+        interactions::Erc20ApproveInteraction,
+        liquidity::slippage::SlippageCalculator,
+        settlement::external_prices::ExternalPrices,
+    },
+};
+
+const DEFAULT_QUOTE_SLIPPAGE_BPS: u32 = 100; // 1%
+
+pub fn encode_interactions(
+    eth: &Ethereum,
+    interactions: &[solution::Interaction],
+) -> Result<Vec<eth::Interaction>, boundary::Error> {
+    let slippage_calculator = SlippageCalculator::from_bps(DEFAULT_QUOTE_SLIPPAGE_BPS, None);
+    let external_prices = ExternalPrices::new(eth.contracts().weth().address(), Default::default())
+        .expect("empty external prices is valid");
+    let slippage_context = slippage_calculator.context(&external_prices);
+
+    let mut settlement = solver::settlement::Settlement::new(Default::default());
+
+    for interaction in interactions {
+        let allowances = interaction.allowances();
+        for allowance in allowances {
```

Comment:
I'm really sorry about leaving this comment but the fact that these are two lines really hurts my eyes
```suggestion
        for allowance in interaction.allowances() {
```